lib/traject/macros/custom.rb

```diff
@@ -99,7 +101,7 @@ def extract_link_data(link_type: 'full')
             link_data << collect_subfield_values(field: field, code: 'u')
           end
           link_data.flatten.compact.each do |link|
-            url_match = regex_split link, %r{https*://([\w*|\.*]*)}
+            url_match = regex_split link, %r{https*://([\w*|.*]*)}
```

Comment:
no need for `\` ?

Reply:
This was fixed automatically by: https://docs.rubocop.org/rubocop/cops_style.html#styleredundantregexpescape

RubyMine warns too:

![Screen Shot 2020-10-26 at 3 48 42 PM](https://user-images.githubusercontent.com/523381/97221014-bee8ca00-17a2-11eb-8600-535824d2eaca.png)

I was unable to find ruby lang docs supporting this issue. I did test this in rubular though and it seems to check out

Reply:
Had no idea escaping. Kinda wild:

```
irb(main):012:0> a = /a[.]c/
=> /a[.]c/
irb(main):013:0> b = /a.c/
=> /a.c/
irb(main):014:0> 'abc'.match? a
=> false
irb(main):015:0> 'abc'.match? b
=> true
irb(main):016:0> 'a.c'.match? a
=> true
irb(main):017:0> 'a.c'.match? b
=> true
irb(main):018:0> c = /a[\.]c/
=> /a[\.]c/
irb(main):019:0> 'abc'.match? c
=> false
irb(main):020:0> 'a.c'.match? c
=> true
```

Also found this, see under ["Metacharacters Inside Character Classes"](https://www.regular-expressions.info/charclass.html)

Reply:
With that knowledge, then there are other flaws with this regex. For example, the `*` is taken as a literal, not a metacharacter of "zero or more of the previous character"

Reply:
oh yes, `*` is a literal in character class. so the below seems to still work? tried with rubular:

`%r{https*://([\w|.]*)}`

Reply:
the pipe is a literal too. that area inside the square brackets treats all characters as "OR"

so it reads like "[any word character OR a pipe OR a dot]" zero or more times following the "//" and capture all that

Reply:
yes that makes sense 👍